### PR TITLE
refactoring request parsing, fixing response format detection

### PR DESCRIPTION
**Pull request by Szasza**
Issue #2 addressed.

### CODE REVIEW

**Comment by MikeRalphson** [comment]
Thanks - looks good. Reminds me I need to set up some tests...

File: index.js [hunk 1]
```diff
@@ -226,12 +235,11 @@ app.get('/api/v1/convert', function(req,res) {
 function finishConversion(res,result,payload){
 	res.set('Access-Control-Allow-Origin','*');
 	res.set('Content-Type',payload.contentType);
-	if (payload.yaml) {
+	if (payload.responseYaml) {
```
Comment (MikeRalphson) [inline index.js]: I think this should be `payload.yamlResponse` ?

Comment (Szasza) [reply index.js]: Absolutely.

**Comment by Szasza** [comment]
Fixed, good catch.

**Comment by MikeRalphson** [comment]
Thanks @Szasza - good fix and improvement!

**Comment by Szasza** [comment]
Thank you and my pleasure.

**Pull request merged**